### PR TITLE
feat: [CDS-59260]: Adding support for quartz expression for scheduled triggers

### PR DESCRIPTION
**Pull request by Y0709**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46904)

### CODE REVIEW

**Review by vcalasansh — DISMISSED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46904)* status: 7 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/beans/entity/NGTriggerEntity.java [hunk 1]
```diff
@@ -134,7 +135,8 @@ public List<Long> recalculateNextIterations(String fieldName, boolean skipMissed
     }
     try {
       String cronExpr = metadata.getCron().getExpression();
-      expandNextIterations(skipMissed, throttled, cronExpr, nextIterations);
+      String cronType = StringUtils.isBlank(metadata.getCron().getType()) ? "UNIX" : metadata.getCron().getType();
+      expandNextIterations(skipMissed, throttled, cronExpr, nextIterations, cronType);
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/beans/entity/NGTriggerEntity.java:139]: nit: Possible to pass enum and check for enum in PersistentNGCronIterable.java.

**Review by sribalij — DISMISSED** [review]

File: 960-persistence/src/main/java/io/harness/iterator/PersistentNGCronIterable.java [hunk 2]
```diff
@@ -45,7 +47,12 @@ default boolean expandNextIterations(boolean skipMissing, long throttled, String
       return removed;
     }
 
-    final Cron cron = parser.parse(cronExpression);
+    final Cron cron;
+    if (cronType.equals("UNIX")) {
```
Comment (sribalij) [inline 960-persistence/src/main/java/io/harness/iterator/PersistentNGCronIterable.java:51]: may be better to ignore case

**Review by sribalij — APPROVED** [review]

File: 960-persistence/src/main/java/io/harness/iterator/PersistentNGCronIterable.java (hunk 2 above)
Comment (Y0709) [reply 960-persistence/src/main/java/io/harness/iterator/PersistentNGCronIterable.java]: done

**Comment by Y0709** [comment]
trigger messagecheck

**Review by vcalasansh — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46904)* status: 7 file(s) to review. 1. unresolved discussions.

**Comment by Y0709** [comment]
trigger review

**Pull request merged**